extensions/hearthstone/src/card-list.tsx

```diff
@@ -0,0 +1,275 @@
+        setCards(collectibleCards.slice(0, INITIAL_PAGE_SIZE));
+        setHasMoreCards(collectibleCards.length > INITIAL_PAGE_SIZE);
+
+        setIsLoading(false);
+      } catch (error) {
+        console.error('Error loading card data:', error);
+        setIsLoading(false);
+      }
+    };
+
+    loadAllCardData();
+  }, [language]);
+
+  // 处理筛选变化
+  useEffect(() => {
+    if (allCardData.length === 0) return;
+
+    setIsLoading(true);
+
+    try {
+      let newFilteredCards = allCardData.filter((card: Card) => card.collectible);
+
+      if (searchTerm) {
+        newFilteredCards = newFilteredCards.filter((card) =>
+          card.name.toLowerCase().includes(searchTerm.toLowerCase()),
+        );
+      }
+
+      if (searchByCost !== 'all') {
+        newFilteredCards = newFilteredCards.filter((card) => {
+          // 处理 card.cost 可能为 undefined 的情况
+          const cost = card.cost ?? 0;
+
+          if (searchByCost === '>10') {
+            return cost > 10;
+          }
+          return cost === parseInt(searchByCost, 10);
+        });
+      }
+
+      if (searchByClass !== 'all') {
+        newFilteredCards = newFilteredCards.filter((card) => {
+          if (searchByClass === 'NEUTRAL') {
+            return card.cardClass === 'NEUTRAL';
+          }
+          return card.cardClass?.toUpperCase() === searchByClass.toUpperCase();
+        });
+      }
+
+      if (searchBySet !== 'all') {
+        newFilteredCards = newFilteredCards.filter((card) => card.set === searchBySet);
+      }
+
+      setFilteredCards(newFilteredCards);
+
+      // 重置可见数量并显示初始卡牌
+      setVisibleCount(INITIAL_PAGE_SIZE);
+      setCards(newFilteredCards.slice(0, INITIAL_PAGE_SIZE));
+      setHasMoreCards(newFilteredCards.length > INITIAL_PAGE_SIZE);
+    } catch (error) {
+      console.error('Error applying filters:', error);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [allCardData, searchTerm, searchByCost, searchByClass, searchBySet]);
+
+  const handleFilterChange = (value: string) => {
+    const parts = value.split('_');
+    const newLanguage = parts[0] as 'enUS' | 'zhCN';
+    const filterType = parts[1];
+    const filterValue = parts.slice(2).join('_');
+
+    if (newLanguage !== language) {
+      setLanguage(newLanguage);
+      return;
+    }
+
+    switch (filterType) {
+      case 'cost':
+        setSearchByCost(filterValue);
+        break;
+      case 'class':
+        setSearchByClass(filterValue);
+        break;
+      case 'set':
+        setSearchBySet(filterValue);
+        break;
+    }
+  };
+
+  // 加载更多卡牌（不关闭窗口）
+  const loadMoreCards = () => {
+    if (hasMoreCards && !isLoading) {
+      const newVisibleCount = visibleCount + LOAD_MORE_SIZE;
+      setVisibleCount(newVisibleCount);
+      setCards(filteredCards.slice(0, newVisibleCount));
+      setHasMoreCards(filteredCards.length > newVisibleCount);
+      showHUD(`已加载 ${Math.min(newVisibleCount, filteredCards.length)}/${filteredCards.length} 张卡牌`);
```

Comment:
The showHUD can be changed to showToast to prevent the main window from closing when loading more cards.